src/lib/middleware/rbac-middleware.ts

```diff
@@ -47,33 +47,10 @@ const rbacMiddleware = (
             let { projectId } = params;
 
             // Temporary workaround to figure out projectId for feature toggle updates.
-            if (permission === DELETE_FEATURE) {
-                const { featureName } = params;
-                projectId = await featureToggleStore.getProjectId(featureName);
-            } else if (permission === UPDATE_FEATURE) {
-                // if projectId of feature is different from project in body
-                // need to check that we have UPDATE_FEATURE access on both old and new project
-                // TODO: Look at this to make it smoother once we get around to looking at project
-                // Changing project of a toggle should most likely be a separate endpoint
+            // will be removed in Unleash v5.0
+            if ([DELETE_FEATURE, UPDATE_FEATURE].includes(permission)) {
                 const { featureName } = params;
                 projectId = await featureToggleStore.getProjectId(featureName);
-                const newProjectId = req.body
-                    ? req.body.project || projectId
-                    : projectId;
-                if (newProjectId !== projectId) {
-                    return (
-                        accessService.hasPermission(
-                            user,
-                            permission,
-                            projectId,
-                        ) &&
-                        accessService.hasPermission(
-                            user,
-                            permission,
-                            newProjectId,
-                        )
-                    );
-                }
```

Comment:
This is a dead code path. Also it will not work because `accessService.hasPermission` is an async function returning a promise.